lib/gitsh/prompter.rb

```diff
@@ -11,25 +10,33 @@ class Prompter
     def initialize(options={})
       @env = options.fetch(:env)
       @use_color = options.fetch(:color, true)
-      @prompt_color = options.fetch(:prompt_color) { PromptColor.new(@env) }
       @options = options
     end
 
     def prompt
-      padded_prompt_format.gsub(/%[bBcdDw#]/, {
-        '%b' => branch_name,
-        "%B" => shortened_branch_name,
-        '%c' => status_color,
-        '%d' => Dir.getwd,
-        '%D' => File.basename(Dir.getwd),
-        '%w' => clear_color,
-        '%#' => terminator
-      })
+      reset_chached_repo_info
```

Comment:
We generally avoid creating state outside of `#initialize` methods in this codebase, and often create private classes that can hold the state they need to process a specific call (e.g. in the `Gitsh::Completer#call` method). Does it make sense to follow that pattern here too?

Something like this maybe?

```ruby
def prompt
    Prompt.new(
      env: env,
      repo_initialized: …,
      branch_name: …,
      repo_status: …,
    ).to_s
end
```